msteams-platform/concepts/device-capabilities/qr-barcode-scanner-capability.md

```diff
@@ -0,0 +1,76 @@
+
+* User aligns a QR or barcode within a frame at the centre of the scanner UI and the code gets scanned automatically. The stored data is displayed instantly in the corresponding information field. This method avoids inconvenience and human-errors of entering the lengthy product code and other relevant information manually. 
+To integrate QR or barcode scanner capability, you must update the app manifest file and call the **`scanBarCode` API**. 
+* The integration allows webapp developers on Teams platform to leverage QR or barcode scanning functionality with Teams JavaScript Client SDK.
+
+For effective integration, you must have a good understanding of [code snippets](#code-snippets) for calling the `scanBarCode` API, which allows you to use native QR or barcode scanner capability. The API throws error in case of an unsupported barcode standard.
+It is important to familiarize yourself with the [API response errors](mobile-camera-image-permissions.md#error-handling) to handle the errors that arise, while calling the API in your Teams app.
+
+> [!NOTE] 
+> Currently, Microsoft Teams support for QR or barcode scanner capability is available for mobile clients only.
+
+## Update manifest
+
+Update your Teams app [manifest.json](../../resources/schema/manifest-schema.md#devicepermissions) file by adding the `devicePermissions` property and specifying `media`. This file update allows your app to ask for requisite permissions from users before they use the QR or barcode scanner for integration.
+
+``` json
+"devicePermissions": [
+    "media",
+],
+```
+
+> [!NOTE]
+> The **Request Permissions** prompt is automatically displayed when a relevant Teams API is initiated. For more information, see [Request device permissions](native-device-permissions.md).
+
+## Scanner capability API
+
+`ScanBarCode` API invokes scanner control that allows the user to scan different types of QR or barcode and returns the scanned result as a string.
+The steps to use QR or barcode scanner capability API are as follows: 
+1. Entry point or invoke: 
+The entry point is the point from where a developer invokes `ScanBarCode` API. This point is a button with the label **Scan QR code**. 
+1. Scanner user experience or execution: This comprises of the scanner experience provided by the API to the user as shown in the following section. The major elements include: 
+    * Camera input stream with an overlay, with a frame at the center. The frame directs the user to align the code as per the provided frame. 
+    * Flash button to toggle flash on or off. 
+    * Close **X** button on top-left to abort scanning operation. 
+1. Exit or return:  
+    * Successfully scanned a barcode 
+    * Scan timed out 
+    * User pressed back button or closed the scanner UI 
+
+**Web app experience for `ScanBarCode` API for QR or barcode scanner capability**
+![web app experience for qr or barcode scanner capability](../../assets/images/tabs/qr-barcode-scanner-capability.png)
+
```

Comment:
Please add error handling section before code snippets. Details:


Error code | Error name | Condition
-- | -- | --
100 | NOT_SUPPORTED_ON_PLATFORM | API is not supported on the current platform.
500 | INTERNAL_ERROR | Internal error is encountered while performing the required operation.
1000 | PERMISSION_DENIED | Permission is denied by the user.
3000 | NO_HW_SUPPORT | Underlying hardware does not support the capability.
4000 | INVALID_ARGUMENTS | One or more arguments are invalid.
8000 | USER_ABORT | User aborts the operation.
8001 | OPERATION_TIMED_OUT | Could not detect the barcode in the given time interval.
9000 | OLD_PLATFORM | Platform code is outdated and does not implement this API.